[bug] panic when no font files found

### What happend?

When running on a linux system without the hardcoded list of `ttf` files, initialising a new MasterWindow will panic the application.

note: cmgui example works as intended on the same system without changes (is it using the imgui embedded fonts?)

The pr [go-findfont/pull/1](https://github.com/AllenDang/go-findfont/pull/1) will add the required paths for NixOS, also fixes `XDG_DATA_HOME`.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x552fbc]

goroutine 1 [running, locked to thread]:
github.com/AllenDang/giu.CreateContext({0xc17318, 0xc00016c000})
        /home/lor/lib/src/github.com/AllenDang/giu/Context.go:82 +0x1fc
github.com/AllenDang/giu.NewMasterWindow({0xbc41f5, 0xe}, 0x280, 0x1e0, 0x0)
        /home/lor/lib/src/github.com/AllenDang/giu/MasterWindow.go:95 +0x1b7
main.main()
        /home/lor/lib/src/github.com/AllenDang/giu/examples/align/align.go:62 +0x2b
exit status 2
```

### Code example

<details><summary><code>main.go</code></summary>

```golang
package main

import (
	g "github.com/AllenDang/giu"
)

func loop() {}
func main() {
	wnd := g.NewMasterWindow("Hello world", 400, 200, 0)
	wnd.Run(loop)
}
```

</details>

<details><summary><code>crashes.log</code></summary>

Using a simple flake development shell here just to have `go` and `gopls` in my $PATH

```
[I] lor@pad ~/l/s/g/A/giu (flake) [1]> go run ./examples/helloworld
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x54de5c]

goroutine 1 [running, locked to thread]:
github.com/AllenDang/giu.CreateContext({0xc08cf8, 0xc0000f0050})
        /home/lor/lib/src/github.com/AllenDang/giu/Context.go:82 +0x1fc
github.com/AllenDang/giu.NewMasterWindow({0xbb6544, 0xb}, 0x190, 0xc8, 0x0)
        /home/lor/lib/src/github.com/AllenDang/giu/MasterWindow.go:95 +0x1b7
main.main()
        /home/lor/lib/src/github.com/AllenDang/giu/examples/helloworld/helloworld.go:17 +0x2b
exit status 2
[I] lor@pad ~/l/s/g/A/giu (flake) [1]> touch ~/.local/share/fonts/FreeSans.ttf
[I] lor@pad ~/l/s/g/A/giu (flake)> go run ./examples/helloworld
panic: Assertion failed!
File: /home/runner/work/cimgui-go/cimgui-go/cimgui/imgui/imgui_draw.cpp, Line 2274

Expression: font_data_size > 100 && "Incorrect value for font_data_size!"

goroutine 6 [running]:
github.com/AllenDang/cimgui-go.init.func1({0xc000036d80?, 0x0?}, {0xc0001c4e10?, 0x0?}, 0x0?)
        /home/lor/lib/pkg/mod/github.com/!allen!dang/cimgui-go@v0.0.0-20240424153022-294abe7370df/assert.go:28 +0x65
github.com/AllenDang/cimgui-go.goImguiAssertHandler(0x0?, 0xc29f30, 0x0?)
        /home/lor/lib/pkg/mod/github.com/!allen!dang/cimgui-go@v0.0.0-20240424153022-294abe7370df/assert.go:45 +0x67
github.com/AllenDang/cimgui-go._Cfunc_ImFontAtlas_AddFontFromFileTTF(0x2245ba0, 0x7f29a4022d50, 0x41700000, 0x7f29a4022cb0, 0x7f29a4022c80)
        _cgo_gotypes.go:5496 +0x4c
github.com/AllenDang/cimgui-go.(*FontAtlas).AddFontFromFileTTFV.func2(0x2245ba0, 0x7f29a4022d50, 0x41700000, 0x7f29a4022cb0, 0x7f29a4022c80)
        /home/lor/lib/pkg/mod/github.com/!allen!dang/cimgui-go@v0.0.0-20240424153022-294abe7370df/cimgui_funcs.go:873 +0x92
github.com/AllenDang/cimgui-go.(*FontAtlas).AddFontFromFileTTFV(0x7f29a4022cb0?, {0xc00001e750?, 0x0?}, 0x41700000, 0xc0000d2bd8, 0x7f29a4022c80)
        /home/lor/lib/pkg/mod/github.com/!allen!dang/cimgui-go@v0.0.0-20240424153022-294abe7370df/cimgui_funcs.go:873 +0xca
github.com/AllenDang/giu.(*FontAtlas).rebuildFontAtlas(0xc0001a2070)
        /home/lor/lib/src/github.com/AllenDang/giu/FontAtlasProsessor.go:301 +0x5ee
github.com/AllenDang/giu.(*MasterWindow).beforeRender(0x0?)
        /home/lor/lib/src/github.com/AllenDang/giu/MasterWindow.go:196 +0x55
github.com/AllenDang/cimgui-go.beforeRender()
        /home/lor/lib/pkg/mod/github.com/!allen!dang/cimgui-go@v0.0.0-20240424153022-294abe7370df/backend.go:40 +0x34
github.com/AllenDang/cimgui-go._Cfunc_igGLFWRunLoop(0x2293170, 0x553360, 0x5533d0, 0x553440, 0x553520)
        _cgo_gotypes.go:24265 +0x45
github.com/AllenDang/cimgui-go.(*GLFWBackend).Run.func1(0x1?)
        /home/lor/lib/pkg/mod/github.com/!allen!dang/cimgui-go@v0.0.0-20240424153022-294abe7370df/glfw_backend.go:249 +0x9c
github.com/AllenDang/cimgui-go.(*GLFWBackend).Run(0x0?, 0x0?)
        /home/lor/lib/pkg/mod/github.com/!allen!dang/cimgui-go@v0.0.0-20240424153022-294abe7370df/glfw_backend.go:249 +0x32
main.main.(*MasterWindow).Run.func1()
        /home/lor/lib/src/github.com/AllenDang/giu/MasterWindow.go:244 +0xcb
github.com/faiface/mainthread.Run.func1()
        /home/lor/lib/pkg/mod/github.com/faiface/mainthread@v0.0.0-20171120011319-8b78f0a41ae3/mainthread.go:37 +0x22
created by github.com/faiface/mainthread.Run in goroutine 1
        /home/lor/lib/pkg/mod/github.com/faiface/mainthread@v0.0.0-20171120011319-8b78f0a41ae3/mainthread.go:36 +0xae
exit status 2
```
</details>

### To Reproduce

1. Remove or hide hardcoded fonts from the system folders
2. run any example or the provided example

### Version

master

### OS

NixOS, other lesser used linux distributions

--- a/Context.go
+++ b/Context.go
@@ -79,7 +79,7 @@ func CreateContext(b imgui.Backend[imgui.GLFWWindowFlags]) *context {
 		fonts := result.IO().Fonts()
 		fonts.AddFontDefault()
 		fontTextureImg, w, h, _ := fonts.GetTextureDataAsRGBA32()
-		tex := Context.backend.CreateTexture(fontTextureImg, int(w), int(h))
+		tex := result.backend.CreateTexture(fontTextureImg, int(w), int(h))
 		fonts.SetTexID(tex)
 		fonts.SetTexReady(true)
 	} else {

